packages/connectivity/src/scp-cf/destination/destination-from-env.ts

```diff
@@ -168,3 +168,47 @@ function destinationAuthToken(
     "Option 'forwardAuthToken' was set on destination but no token was provided to forward. This is most likely unintended and will lead to a authorization error on request execution."
   );
 }
+
+/**
+ * @internal
+ */
+export function validateNameAvailable(
+  destinationName: string,
+  existingNames: Set<string>
+): void {
+  if (existingNames.has(destinationName)) {
+    throw new Error(
+      `Parsing mocked destinations failed, destination with name "${destinationName}" already exists in the "destinations" environment variables.`
+    );
+  }
+}
+
+/**
+ * @internal
+ */
+export function setDestinationsInEnv(destinations: Destination[]): void {
+  process.env.destinations = JSON.stringify(destinations);
+}
+
+/**
+ * Set a given destination in the `destinations` environment variable.
+ *
+ * Throws an error if a destination with the same name as the given test destination already exists.
+ * @param destination - Test destination to add to the `destinations` environment variable
+ */
+export function registerDestination(destination: Destination): void {
+  const currentDestinations = getDestinationsFromEnv();
+  const existingNames = new Set<string>(
+    currentDestinations.map(dest => {
+      if (!dest.name) {
```

Comment:
Why this is a problem? I think our set method should ensure that you can not set a destination with undefined name, but if they are in the env by some mechanism is this a problem? Also without some context you get an exception which you can not investigate. So perhaps use a warning and give some info on the destination. Be careful not to expose sensitve information in the log. So perhaps it is easier to just get rid of this exception?